cmd/ip.go

```diff
@@ -73,89 +69,89 @@ func (s ipManager) getAllIPs() []net.IP {
 	return ips
 }
 
-func (s ipv6Stack) filterIPs(ips []net.IP) []net.IP {
-	// ip.IsGlobalUnicast() returns true for:
-	// GUA = Global Unicast Address
-	// ULA = Unique Local Address
-	// We prefer GUA over ULA.
-	ipv6s := []net.IP{}
-	for _, ip := range ips {
-		if ip.IsGlobalUnicast() && ip.To4() == nil {
-			ipv6s = append(ipv6s, ip)
-		}
-	}
+func (s ipv6Stack) getBaseScore(ip net.IP) uint16 {
+	score := uint16(1)
 
-	return ipv6s
-}
+	if ip.To4() != nil {
+		return 0
+	}
+	if !ip.IsGlobalUnicast() {
+		return 0
+	}
 
-func (s ipv4Stack) filterIPs(ips []net.IP) []net.IP {
-	ipv4s := []net.IP{}
-	for _, ip := range ips {
-		if ip.To4() != nil {
-			ipv4s = append(ipv4s, ip)
+	if ipv6IsGUA(ip) {
+		score += 0x8
+		if ipv6IsEUI64(ip) {
+			score += 0x4
 		}
 	}
```

Comment:
From [RFC-4193](https://datatracker.ietf.org/doc/html/rfc4193#section-3.2.2) and [RFC-4291](https://datatracker.ietf.org/doc/html/rfc4291#section-2.5.1), it seems like both Global Unicast addresses (GUA) and Unique Local addresses (ULA) can have the interface ID part in the modified EUI-64 format. So these score points do not depend on each other.

```suggestion
	if ipv6IsGUA(ip) {
		score += 0x8
	}

	if ipv6IsEUI64(ip) {
		score += 0x4
	}
```